cmake/StlabUtil.cmake

```diff
@@ -146,18 +146,20 @@ endfunction()
 # |--------------+--------------------------------------------|
 # | libdispatch  | libdispatch (aka. Grand Central Dispatch ) |
 # | emscripten   | Emscripten's executor                      |
-# | qt           | Qt's event framework                       |
+# | qt5          | Qt's event framework (Qt5)                 |
+# | qt6          | Qt's event framework (Qt6)                 |
 # | none         | None                                       |
 function( stlab_detect_main_executor result_var )
-  find_package(Qt6 QUIET COMPONENTS Core)
   stlab_detect_task_system( task_system )
 
   if( task_system STREQUAL "libdispatch" )
     set( result "libdispatch")
   elseif( CMAKE_SYSTEM_NAME STREQUAL "Emscripten" )
     set( result "emscripten")
-  elseif( Qt6Core_FOUND )
-    set( result "qt")
+  elseif( TARGET Qt6::Core )
+    set( result "qt6")
+  elseif( TARGET Qt5::Core )
+    set( result "qt5")
```

Comment:
Projects injecting their own Qt targets for in-source builds or other reasons need to implement the contract set by the corresponding CMake modules (in this case they'd need to match what is documented [here](https://doc.qt.io/qt-6/cmake-variable-reference.html)). I don't know how one could reason about this otherwise. This is a correctness issue IMO.

Reply:
The thing is, implementing the full interface of every find modules is very hard and time consuming, and it is often complicated by the fact that some projects have their own find modules with their own custom interface :confused:. I agree with you that anyone implementing a custom `FindQt` _should_ define every single variable defined by the Qt cmake files, but if we can easily remove some of this burden, why not? "Be conservative in what you do, be liberal in what you accept from others" :slightly_smiling_face:.

Most sources I'm aware of recommend relying on targets rather than variables (as a matter of fact, the Qt doc you linked hints at that as well:
> Note: You rarely need to access these variables directly. Common tasks like linking against a module should be done through the library targets each module defines.

). In addition to being more self-contained than variables, they are less amenable to mutation and can be scoped globally (for instance, `find_package(GLOBAL)` will make imported targets available to the entire project but not variables).

I've made the change to `Qt5_FOUND` and `Qt6_FOUND` as requested but I still think that the version that checks the targets existence would be nicer for projects integrating stlab.

Reply:
The Qt comment refers to variables like `_LIBRARIES` which were historically used before targets were a feature. That is different...

If a `find_package(Foo)` has `Foo_FOUND` set to `FALSE`, there is no guarantee that a broken `Foo::Foo` was not created. The contract guarantees certain entities when `Foo_FOUND` is `TRUE` but there could be other ones set as well and there are no guarantees at all when `FALSE`.